CDP4WebServices.API/Services/Operations/SideEffects/Implementation/OptionSideEffect.cs

```diff
@@ -142,6 +151,83 @@ public override void BeforeDelete(Option thing, Thing container, NpgsqlTransacti
             {
                 throw new InvalidOperationException($"Cannot delete the only option with id {thing.Iid}.");
             }
+
+            if (container is Iteration iteration)
```

Comment:
I am not sure if @mmilitaru mentioned it to you but he proposed that this functionality should be in `AfterDelete` as in only set the default option to null after the option has been deleted.